### DISCUSSION TITLE
[Crashlytics] 'Crashlytics_Platform.h' file not found (macos)

### DISCUSSION THREAD

**Discussion by kostasoft** [firebase/flutterfire#7484] [Q&A]
The project is normally built for ios, but for macos it gives the following error.
I am laying out a screenshot from Xcode, it is well visible, which error and in which file appears.

The file **pubspec.yaml** contains the following code
```
...
  firebase_core: ^1.10.2
  firebase_messaging: ^11.2.0
  firebase_crashlytics: ^2.4.0
...
```

<img width="1440" alt="Снимок экрана 2021-12-07 в 17 44 35" src="https://user-images.githubusercontent.com/7292854/145051670-07cbd222-0247-4216-932f-2f6061b75c9d.png">

Again, the problem is only with the build of the project for MacOs.

**Comment by a-v-ebrahimi** [comment]
Same as here
Annoying I can't compile for MacOS, but I can compile for iOS and Android :|

**Comment by a-v-ebrahimi** [comment]
I created an issue for it and @Salakar assigned it to himself:
https://github.com/FirebaseExtended/flutterfire/issues/7558

**Comment by Alverd04** [comment]
Same issue tried everything!

**Reply by a-v-ebrahimi** [reply]
Temporary work around:  Just copy **Crashlytics_Platform.h** from iOS pod folder to macOS folder!

**Reply by a-v-ebrahimi** [reply]
![Screen Shot 2022-01-09 at 9 47 29 PM](https://user-images.githubusercontent.com/179375/148695148-0bac799a-c2d6-426b-9c4d-badb8c1f0196.png)

**Reply by mattsellars** [reply]
This worked for me. In terminal I simply ran `cp $HOME/developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-2.4.5/ios/Classes/Crashlytics_Platform.h $HOME/developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-2.4.5/macos/Classes/Crashlytics_Platform.h` and I will add it to my Runner xcproject pre build script as a work around for now.

**Reply by mattsellars** [reply]
This worked adding it as a pre build script in the Xcode runner.

Enter the scheme editor and add a "Pre-action script". Xcode UI looks like the following:

_(Note: make sure you select your target in the "provide build settings from" drop down)_
<img width="931" alt="Screen Shot 2022-01-14 at 4 36 25 PM" src="https://user-images.githubusercontent.com/236337/149598441-8eaeca48-4a93-491a-90d1-f2d5f9c3cd96.png">


In VSCode running the app now outputs:

```
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id: <redacted> }
{ platform:macOS, name:Any Mac }
<pushd output>
Running Crashlytics version 2.4.5 header fix for macOS
<popd output>
```

Updated my script as follows to attempt to extract the crashlytics version ymmv:



```
pushd $SRCROOT/..
#echo "Current dir:"
#pwd
CRASHLYTICS_VERSION=`flutter pub deps | grep "firebase_crashlytics " | sed "s/.*firebase_crashlytics //g"`
#trim whitespace
CRASHLYTICS_VERSION=`echo $CRASHLYTICS_VERSION | sed -e 's/^[[:space:]]*//'`
echo "Running Crashlytics version $CRASHLYTICS_VERSION header fix for macOS"

cp $HOME/developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-$CRASHLYTICS_VERSION/ios/Classes/Crashlytics_Platform.h $HOME/developer/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_crashlytics-$CRASHLYTICS_VERSION/macos/Classes/Crashlytics_Platform.h
popd
```

**Reply by mattsellars** [reply]
Note if it can't find flutter in bash you might need export your path to it like `export PATH="$PATH:~/Developer/flutter/bin"`
